File: src/c#/main/commands/RichJson_merge_folder.cs
using System;
using System.Collections.Generic;
using main.core;
using main.helper;

namespace main.commands
{
    public class RichJson_merge_folder
    {
        public object Execute(RichJsonParser parser, RichJsonContext context)
        {
            var folderContent = (IDictionary<string, object>)RichJsonCommandHolder.ExecuteCommand("folder", parser, context);
            List<string> sortedKeys = RichJsonHelper.GetKeysSorted(folderContent);
            var result = new Dictionary<string, object>();

            if (sortedKeys.Count > 0)
            {
                foreach (string key in sortedKeys)
                {
                    object content = folderContent[key];
                    if (content is IDictionary<string, object> contentMap)
                    {
                        RichJsonHelper.MergeIntoTarget(result, contentMap);
                    }
                }
            }

            return result;
        }
    }
}

File: src/c++/main/commands/RichJson_merge_folder.cpp
#include "RichJson_merge_folder.hpp"
#include "../core/RichJsonCommandHolder.hpp"
#include "../core/RichJsonContext.hpp"
#include "../helper/RichJsonHelper.hpp"

namespace RichJson {

json richJson_merge_folder(RichJsonParser& parser, RichJsonContext& context) {
    json folderContent = RichJsonCommandHolder::executeCommand("folder", parser, context);
    auto sortedKeys = RichJsonHelper::getKeysSorted(folderContent);

    json result = json::object();
    for (const auto& key : sortedKeys) {
        const json& content = folderContent.at(key);
        if (content.is_object()) {
            result = RichJsonHelper::mergeIntoTarget(result, content);
        }
    }
    return result;
}

} // namespace RichJson
